tests/test_validators_dataclass.py

```diff
@@ -53,20 +53,50 @@ class MyDataclass:
         @classmethod
         def check_a_and_b(cls, v, info):
             if len(v) < 4:
-                raise TypeError(f'{info.field_name} is too short')
+                raise ValueError(f'{info.field_name} is too short')
             return v + 'x'
 
     assert asdict(MyDataclass(a='1234', b='5678')) == {'a': '1234x', 'b': '5678x'}
 
     with pytest.raises(ValidationError) as exc_info:
         MyDataclass(a='x', b='x')
     assert exc_info.value.errors() == [
-        {'loc': ('a',), 'msg': 'a is too short', 'type': 'type_error'},
-        {'loc': ('b',), 'msg': 'b is too short', 'type': 'type_error'},
+        {
+            'ctx': {'error': 'a is too short'},
+            'input': 'x',
+            'loc': ('a',),
+            'msg': 'Value error, a is too short',
+            'type': 'value_error',
+        },
+        {
+            'ctx': {'error': 'b is too short'},
+            'input': 'x',
+            'loc': ('b',),
+            'msg': 'Value error, b is too short',
+            'type': 'value_error',
+        },
     ]
 
 
-# @pytest.mark.xfail(reason='working on V2')
+def test_type_error():
+    @dataclass
+    class MyDataclass:
+        a: str
+        b: str
+
+        @field_validator('a', 'b')
+        @classmethod
```

Comment:
This isn't necessary. But I think we should, eventually, add it everywhere and pretend as if it is to encourage users to do the same. But we should do that, and just that, in a future PR to avoid polluting diffs. But this is fine here for now, no need to change it.